core/src/banking_stage.rs

```diff
@@ -955,43 +953,34 @@ impl BankingStage {
         let mut slot_metrics_tracker = LeaderSlotMetricsTracker::new(id);
         loop {
             let my_pubkey = cluster_info.id();
-            while !buffered_packet_batches.is_empty() {
-                let (decision, process_buffered_packets_time) = Measure::this(
-                    |_| {
-                        Self::process_buffered_packets(
-                            &my_pubkey,
-                            &socket,
-                            poh_recorder,
-                            cluster_info,
-                            &mut buffered_packet_batches,
-                            &forward_option,
-                            transaction_status_sender.clone(),
-                            &gossip_vote_sender,
-                            &banking_stage_stats,
-                            &recorder,
-                            data_budget,
-                            &qos_service,
-                            &mut slot_metrics_tracker,
-                        )
-                    },
-                    (),
-                    "process_buffered_packets",
-                );
-                slot_metrics_tracker
-                    .increment_process_buffered_packets_us(process_buffered_packets_time.as_us());
-
-                if matches!(decision, BufferedPacketsDecision::Hold)
-                    || matches!(decision, BufferedPacketsDecision::ForwardAndHold)
-                {
-                    // If we are waiting on a new bank,
-                    // check the receiver for more transactions/for exiting
-                    break;
-                }
-            }
+            let (_, process_buffered_packets_time) = Measure::this(
+                |_| {
+                    Self::process_buffered_packets(
+                        &my_pubkey,
+                        &socket,
+                        poh_recorder,
+                        cluster_info,
+                        &mut buffered_packet_batches,
+                        &forward_option,
+                        transaction_status_sender.clone(),
+                        &gossip_vote_sender,
+                        &banking_stage_stats,
+                        &recorder,
+                        data_budget,
+                        &qos_service,
+                        &mut slot_metrics_tracker,
+                    )
+                },
+                (),
+                "process_buffered_packets",
+            );
+            slot_metrics_tracker
+                .increment_process_buffered_packets_us(process_buffered_packets_time.as_us());
 
             let (_, slot_metrics_checker_check_slot_boundary_time) = Measure::this(
                 |_| {
                     let current_poh_bank = {
+                        // TODO (B): constant spinning here may have an impact on poh, discuss in PR
```

Comment:
something worth discussing

Reply:
if you have one packet in buffered_packet_batches that exceeds cost model, it'll probably spin super fast grabbing this lock a ton, so might be good idea to rate limit it

Reply:
yeah this is a good point, we can wrap it in a check that only checks this once every `SLOT_BOUNDARY_CHECK_MS: usize = 10`